src/pybind/mgr/dashboard/frontend/src/app/shared/api/rgw-bucket.service.spec.ts

```diff
@@ -1,21 +1,137 @@
-import { HttpClientModule } from '@angular/common/http';
+import { HttpClient } from '@angular/common/http';
 import { HttpClientTestingModule } from '@angular/common/http/testing';
-import { inject, TestBed } from '@angular/core/testing';
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
+
+import 'rxjs/add/observable/of';
+import 'rxjs/add/operator/mergeMap';
+import { Observable } from 'rxjs/Observable';
 
 import { RgwBucketService } from './rgw-bucket.service';
 
 describe('RgwBucketService', () => {
+  let count = 0;
+  let currentPath: string;
+  let httpClient: HttpClient;
+  let service: RgwBucketService;
+
+  const fake = (path) => {
+    currentPath = path;
+    count++;
+    return Observable.of([]);
+  };
+
   beforeEach(() => {
     TestBed.configureTestingModule({
       providers: [RgwBucketService],
-      imports: [HttpClientTestingModule, HttpClientModule]
+      imports: [HttpClientTestingModule]
     });
+
+    count = 0;
+    currentPath = '';
+    httpClient = TestBed.get(HttpClient);
+    service = TestBed.get(RgwBucketService);
+
+    spyOn(httpClient, 'get').and.callFake(fake);
+    spyOn(httpClient, 'post').and.callFake(fake);
+    spyOn(httpClient, 'put').and.callFake(fake);
+    spyOn(httpClient, 'delete').and.callFake(fake);
   });
 
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it(
+    'should call list',
```

Comment:
This could be more descriptive or split in two tests as the list is called twice. Here you can remove the expectation if it's an observable as you will subscribe to them after you get them - which would fail if they weren't observables.